aspnetcore/fundamentals/configuration/index.md

```diff
@@ -57,6 +57,35 @@ The following code displays the enabled configuration providers in the order the
 
 [!code-csharp[](index/samples/6.x/ConfigSample/Pages/Index2.cshtml.cs?name=snippet)]
 
+## Host configuration
+
+The following list contains host configuration sources from lowest to highest priority:
+
+1. `DOTNET_` prefixed environment variables
+1. Command-line arguments
+1. `ASPNETCORE_` prefixed environment variable
+
+When a configuration value is set in host and application configuration, the host configuration is used.
+
+See [**Explanation** in this GitHub comment](https://github.com/dotnet/AspNetCore.Docs/issues/25626#issuecomment-1098616664) for an explanation of why in host configuration, `ASPNETCORE_` prefixed environment variables have higher priority than command-line arguments.
+
+### Host variables
+
+The following variables are locked in early when initializing the host builders and can't be influenced by application config:
+
+* Application name
+* Environment name, for example `Development`, `Production`, and `Staging`
+* Content root
+* Web root
+* Whether to scan for [hosting startup assemblies](xref:fundamentals/configuration/platform-specific-configuration) and which assemblies to scan for.
+* Variables read by app and library code from [HostBuilderContext.Configuration](xref:Microsoft.Extensions.Hosting.HostBuilderContext.Configuration) in [IHostBuilder.ConfigureAppConfiguration](xref:Microsoft.Extensions.Hosting.IHostBuilder.ConfigureAppConfiguration%2A) callbacks.
+
+`URLS` is notably absent from the previous list of host variables. `URLS` is read later from application config where non-prefixed environment variables override `appsettings.json`. `ASPNETCORE_` prefixed and `DOTNET_` prefixed environment variables set in application configuration are overridden by those set in host configuration.
```

Comment:
I feel like this introduces a contradiction with statement above: `When a configuration value is set in host and application configuration, the host configuration is used.`

Following that statement, my understanding is that `ASPNETCORE_URLS` has highest priority and if `Urls` is provided in application configuration, then `host configuration is used`. But then this section says:
> `ASPNETCORE_` prefixed and `DOTNET_` prefixed environment variables set in application configuration are overridden by those set in host configuration

Which states the opposite. That it is not `host configuration is used`, but rather application configuration is used.

Or is this purely for `Host variables`? But in that case this statement makes me wonder why mention `application configuration` at all if it can't influence host configuration at all:

> The following variables are locked in early when initializing the host builders and can't be influenced by application config:

Reply:
I re-read the statement, and I suppose it states that variables that would be set in application configuration will not be used when host configuration contains those same variables, as application configuration is overridden by host configuration. Which ultimately leads back to overall meaning that host configuration takes precedence over application configuration. Combining statement above that host configuration is used over application configuration and that `ASPNETCORE_ prefixed environment variable` has highest priority, it would make me believe that `ASPNETCORE_URLS` is the highest priority variable.

Reply:
Here's what I wrote in the comment:

> `URLS` is notably absent from this list of host variables. URLS is read later from application config where non-prefixed environment variables override appsettings.json, but `ASPNETCORE_`-prefixed and `DOTNET_`-prefixed environment variables get overridden.

The wording I had here wasn't clear, but I meant to indicate that `URLS` gets overridden by application config, not hosting config, because it is not a "host variable".

I think a lot of settings like `URLS` and `shutdownTimeoutSeconds` feel like they should be "host variables", but they are not by this definition. What if we refer to everything above like content root and environment name as "host bootstrap settings" instead of "host variables"? This would indicate that only settings needed in the very early stages of building the host like content root and environment name are included but not all host-related settings.

How about this?

```suggestion
Every other host setting is read from application config instead of host config.

`URLS` is one of the many common host settings that is not a bootstrap setting. Like every other host setting not in the previous list, `URLS` is read later from application config. Host config is a fallback for application config, so host config can be used to set `URLS`, but it will be overridden by any configuration source in application config like `appsettings.json`.
```